src/cocoa/toga_cocoa/widgets/table.py

```diff
@@ -163,3 +162,32 @@ def scroll_to_row(self, row):
     def rehint(self):
         self.interface.intrinsic.width = at_least(self.interface.MIN_WIDTH)
         self.interface.intrinsic.height = at_least(self.interface.MIN_HEIGHT)
+
+    def _add_column(self, heading, accessor):
+        column_identifier = at(accessor)
+        self.column_identifiers[id(column_identifier)] = accessor
+        column = NSTableColumn.alloc().initWithIdentifier(column_identifier)
+        self.table.addTableColumn(column)
+        self.columns.append(column)
+
+        cell = TogaIconCell.alloc().init()
+        column.dataCell = cell
+
+        column.headerCell.stringValue = heading
+
+    def add_column(self, heading, accessor):
+        self._add_column(heading, accessor)
+        self.table.sizeToFit()
+
+    def remove_column(self, accessor):
+        column = self.table.tableColumnWithIdentifier(accessor)
+        self.table.removeTableColumn(column)
+
+        # delete column and identifier
+        self.columns.remove(column)
+        for k, v in self.column_identifiers.items():
+            if v == accessor:
+                del self.column_identifiers[k]
+                break
```

Comment:
The fact that this iteration dance is required suggests that we're indexing the wrong thing. Should this column_identifiers be a mapping of accessor to column_identifier, rather than the other way around?

Reply:
I changed the direction of the mapping and what's mapping. Now it is mapping:
```python
{ accessor: at(accesor) }
```

Before was
```python
{ id(at(accessor)): accessor}
```

This is useful for reusing it in remove_column.